detect/region.go

```diff
@@ -0,0 +1,66 @@
+package detect
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/urfave/cli"
+	"gopkg.in/errgo.v1"
+
+	"github.com/Scalingo/cli/config"
+	"github.com/Scalingo/cli/utils"
+
+	"github.com/Scalingo/go-scalingo/v4/debug"
+)
+
+// CurrentRegion extract and returns the region from the Git remote URL
+// If not found it return an empty string
+func CurrentRegion(c *cli.Context, rc *config.RegionsCache) string {
+	if dir, ok := utils.DetectGit(); ok {
+		currentRegion, err := getRegionFromGitRemote(dir, c.GlobalString("remote"), rc)
+		if err != nil {
+			debug.Println(err)
+		}
+		debug.Println("[detect] Region name is", currentRegion)
+		return currentRegion
+	}
+	return ""
+}
+
+// ScalingoRepo searches into the current directory and its parent for a remote
+// named remoteName or scalingo-<remoteName>.
+//
+// It returns the region and an error.
+func getRegionFromGitRemote(directory string, remoteName string, rc *config.RegionsCache) (string, error) {
+	remotes, err := utils.ScalingoRemotes(directory)
+	if err != nil {
+		return "", err
+	}
+
+	altRemoteName := "scalingo-" + remoteName
+	for _, remote := range remotes {
+		if remote.Config().Name == remoteName || remote.Config().Name == altRemoteName {
+			return extractRegionFromGitRemote(remote.Config().URLs[0], rc)
+		}
+	}
+
+	return "", errgo.Newf("Scalingo Git remote hasn't been found")
```

Comment:
```suggestion
	return "", errgo.Newf("scalingo Git remote hasn't been found")
```

Error messages should always start with a lower case letter.

Reply:
I will fix that, thanks for the review